### ISSUE TITLE
Formatter undocumented deviation: 

### ISSUE THREAD

**Issue by henribru** [astral-sh/ruff#8179]
[Black formatting](https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4AEtAIddAD2IimZxl1N_WlbvK5V8G17R4xukEnqOHRjoYorGA3_x5d79TklUYR_XtF6MM4d5srbPV3-N4pXQ1SwF_YZZxkksYNb5k5P7chlg8SSpdJuccpW53I3EE2uXGr5zBl09V3WMIRMwonXPxdgmzF3oapEcKjKrbacGLJ8ijg5fK84E8SAp_bjGAAAA8DH6Gx1SCp8AAaMBrgIAAHUkjc-xxGf7AgAAAAAEWVo=):
```
def a():
    return b(
        c,
        d,
        e,
        f=lambda self, *args, **kwargs: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(
            *args, **kwargs
        ),
    )
```

[Ruff formatting](https://play.ruff.rs/?secondary=Format):
```
def a():
    return b(
        c,
        d,
        e,
        f=lambda self,
        *args,
        **kwargs: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(*args, **kwargs),
    )
```

In addition to being incompatible, this formatting seems very strange and not very readable, as the `*args` and `**kwargs` suddenly look like they are arguments to `b`.

**Comment by charliermarsh** [comment]
Thanks!

**Comment by henribru** [comment]
This seems possibly related

https://black.vercel.app/?version=stable&state=_Td6WFoAAATm1rRGAgAhARYAAAB0L-Wj4ADqAHRdAD2IimZxl1N_WlXnON2nzNX9HFvHx6cBUQ15-GcjjCXTWY4lAb3LPNtN0kban15sEExCiEMHC1fqLMN_I2mNYrKs6DX7rL7K_IqhZ4f1qfF_pwoKJfLbTWpcKTtsPNouVDJpGY10D5nlGuf_NfXkJrKjOtjHACUnbTAaUKR-AAGQAesBAAAgR-_jscRn-wIAAAAABFla

https://play.ruff.rs/c9b4e3fa-4401-4248-a98d-d9ae6b984697?secondary=Format

**Comment by charliermarsh** [comment]
Yeah, looks like we have some refinements to make in the lambda formatting. On it! (Thanks for sharing playground links, it helps a lot.)

**Comment by DarthMuzammil** [comment]
@charliermarsh Can i take this issue up?

**Comment by MichaReiser** [comment]
@DarthMuzammil sure. Let me know if you need any help

**Comment by charliermarsh** [comment]
@DarthMuzammil -- have you had a chance to look into this? If not, do you still plan to? (No pressure, I just want to make sure we find an owner soon if it's not being addressed.)

**Comment by MichaReiser** [comment]
It seems that we shouldn't break lambda parameters:

```
def a():
    return lambda self, *args, **kwargs: aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(
        *args, **kwargs
    )
```

**Comment by MichaReiser** [comment]
It seems black never inserts soft line breaks between parameters. This would be an easy fix... except for comments

**Comment by DarthMuzammil** [comment]
@charliermarsh  Hey, sorry for the delay i just saw this message, yes i will start working on this today

**Comment by DarthMuzammil** [comment]
> It seems black never inserts soft line breaks between parameters. This would be an easy fix... except for comments
> 
> Although Black's formatting also has its shortcomings when you have default values:
> 
> ```python
> def a():
>     return b(
>         c,
>         d,
>         e,
>         f=lambda self, araa, kkkwargs=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa(
>             *args, **kwargs
>         ), e=1, f=2, g=2: d,
>     )
> ```
> 
> It could be nice to simply parenthesize the lambda expression and indent it by 1 level (and otherwise keep our formatting). However, that would also mean that the formatter must remove the parentheses again when the lambda fits on a single line—something we haven't really done before in sub-expression positions. And there's also no real good place to add it, because this should be done when formatting lambas in any position.

@MichaReiser How should i go about solving this issue?

**Comment by MichaReiser** [comment]
@DarthMuzammil not sure. I think coming up with a style that works with the presence of comments (which black does not always) is the main work of this issue. 

Related black issue: https://github.com/psf/black/issues/809

**Comment by MichaReiser** [comment]
@ntBre just realized. We shouldl update our black deviation documentation and explain our new lambda formatting (or update the existing section).
